cmd/plume/prerelease.go

```diff
@@ -359,7 +476,33 @@ func azurePreRelease(ctx context.Context, client *http.Client, src *storage.Buck
 	return nil
 }
 
-func awsUploadToPartition(spec *channelSpec, part *awsPartitionSpec, imageName, imageDescription, imagePath string) (map[string]string, map[string]string, error) {
+func getSpecAWSImageName(spec *channelSpec, imageMetadata *map[string]interface{}) (map[string]string, error) {
+	imageFileName := spec.AWS.Image
+	imageName := fmt.Sprintf("%v-%v-%v", spec.AWS.BaseName, specChannel, specVersion)
+	imageName = regexp.MustCompile(`[^A-Za-z0-9()\\./_-]`).ReplaceAllLiteralString(imageName, "_")
+	imageDescription := fmt.Sprintf("%v %v %v", spec.AWS.BaseDescription, specChannel, specVersion)
+
+	if spec.System == "Fedora" {
+		t := template.Must(template.New("filename").Parse(imageFileName))
+		buffer := &bytes.Buffer{}
+		if err := t.Execute(buffer, imageMetadata); err != nil {
+			return nil, err
+		}
+		imageFileName = buffer.String()
+		imageName = strings.TrimSuffix(imageFileName, ".raw.xz")
+		imageDescription = fmt.Sprintf("%v %v %v", spec.AWS.BaseDescription, specChannel, specVersion)
+	}
+
+	awsImageMeta := map[string]string{
```

Comment:
This should be returning a struct, not a map.

Reply:
Changed this to `struct`

Reply:
is this resolved?